### DISCUSSION TITLE
图计算模块支持增量构建和增量计算么

### DISCUSSION THREAD

**Discussion by bookug** [alibaba/GraphScope#1090] [Q&A] [answered]
实践中容易有这个需求。
樊文飞老师做过一系列增量计算的研究。

**Comment by yecol** [comment, marked as answer]
Hi bookug, 

Thanks for your interest in our project! The analytical engine of GraphScope originated from GRAPE, which takes incremental evaluation as a step to parallelize sequential algorithms. In the GRAPE repo, we also provide an example to demonstrate how it supports algorithms run on dynamic graphs(append-only). Please refer to these to get more info:

- [analytical-engine](https://github.com/alibaba/GraphScope/tree/main/analytical_engine)
- [example-on-dynamic-graph(append-only)](https://github.com/alibaba/libgrape-lite/tree/master/examples/gnn_sampler#sampling-on-dynamic-graph-append-only)
